src/scene_battle_rpg2k3.cpp

```diff
@@ -532,6 +532,15 @@ bool Scene_Battle_Rpg2k3::ProcessBattleAction(Game_BattleAlgorithm::AlgorithmBas
 
 		if (action->IsFirstAttack() && action->GetStartSe()) {
 			Game_System::SePlay(*action->GetStartSe());
+			int damageTaken = action->GetSource()->ApplyConditions();
+			if (damageTaken != 0) {
+				DrawFloatText(
+					action->GetTarget()->GetBattleX(),
+					action->GetTarget()->GetBattleY(),
+					0,
+					boost::lexical_cast<std::string>(damageTaken),
+					30);
+			}
 		}
```

Comment:
Everytime anybody does his turn all allies and enemies get state heal/damage. So should be something like this:
```
std::vector<Game_Battler*> battlers;
Main_Data::game_party->GetBattlers(battlers);
Main_Data::game_enemyparty->GetBattlers(battlers);

for (auto& b : battlers) {
  // Your damageTaken code
}
```